### ISSUE TITLE
Upload with tags with tag with a hyphen symbol creates multiple tags

### ISSUE THREAD

**Issue by AevumDecessus** [manyfold3d/manyfold#2704]
**Describe the bug**
If you tag a model in the upload dialog with a tag containing a hyphen character, it creates the tags as expected, but **also** creates a tag with the hyphen character replaced with a space character

**To Reproduce**
1. Add models to the upload dialog
2. Tag the models, with at least one tag having a `-` symbol in it
3. Finish uploading the model
4. 
**Expected behavior**
The model to be uploaded, and tagged as tagged in the upload dialog

**Screenshots**
Tags in upload dialog
![image](https://github.com/user-attachments/assets/a8868230-ae04-4f32-8229-1c6ee218a51c)

Tags applied to models
![image](https://github.com/user-attachments/assets/9ca45b01-51e0-420e-a067-ff72bdc5d126)

**Additional context**
The `double-sided` tag in the upload dialog created a `double-sided` tag as expected, but also a `double sided` tag that is applied to most of the models

**Comment by AevumDecessus** [comment]
This is on the v0.79.0 linuxserver image

**Comment by Floppy** [comment]
Ooh! Interesting! Does it do the same thing in the general editing page?

**Comment by Floppy** [comment]
I checked; it doesn't.

**Comment by Floppy** [comment]
OK, I think I understand why this is happening. When the model is created, it's given the right set of tags, but when it's then scanned, and path template parsing is enabled, it will read the folder names back in and apply them as tags. That should normally be fine, because the folder was created because of the original tag, but in the case where there's a hyphen, the default folder-to-tag behaviour is to remove the hyphen when creating a tag.

This is because both of the tags `hello-world` and `hello world` will convert to `hello-world` on disk.

Hmmm. This has similar problems to capitalisation of tags and folder name safety, in #3735 and #3742.

**Comment by AevumDecessus** [comment]
good to know, makes sense now that you've explained it, fairly trivial to just avoid hyphens in tag names to not run into this

**Comment by josephshanak** [comment]
> good to know, makes sense now that you've explained it, fairly trivial to just avoid hyphens in tag names to not run into this

I don't think this is an acceptable solution. Personally, I want all of my tags to be hyphenated and *not* have spaces so the when organized on disk they do not contain spaces.

**Comment by Floppy** [comment]
I definitely do need to fix this somehow, I agree!

Perhaps the right solution is to just not change hyphens to spaces in tags when found on disk. It's trying to be too clever.

**Comment by Tarakesh83** [comment]
same happens to folders/tags with an underscore in them.
Example from Thingiverse: https://www.thingiverse.com/thing:2695922
the promlematic tag there is warhammer_40k
path gets converted to 
miniature/warhammer/warhammer_40k/40k/tabletop/miniatures/dd/mini/3dprintable/model/painting/wargames/wh40k/zooraw/hanging-shelve-for-up-to-40mm-miniatures#1492

warhammer_40k gets converted to warhammer 40k which then gets readded as new tag

wants to update to new path this time with a hyphen in it.:
miniature/warhammer/warhammer_40k/40k/tabletop/miniatures/wargames/warhammer-40k/dd/mini/wh40k/3dprintable/model/painting/zooraw/hanging-shelve-for-up-to-40mm-miniatures#1492.
